index.ts

```diff
@@ -237,7 +248,7 @@ export default function sveltePlugin(options?: esbuildSvelteOptions): Plugin {
 
                     const result: OnLoadResult = {
                         contents,
-                        warnings: warnings.map(convertMessage),
+                        warnings: warnings.filter(options?.filterWarnings).map(convertMessage),
```

Comment:
TS complains here / wants you to add the `?` because control flow is reset inside functions. TS can't know for sure that there isn't something else that resets this options to undefined between you setting it to sth and this function getting called.

Reply:
I'm not questioning TSs decision here, but there is only a single assignment to `options` in that entire file. I _know_ it will never be undefined :shrug:

Reply:
You can do `options!.filterWarnings` in that case